Two-body-scf-guess.py

```diff
@@ -0,0 +1,157 @@
+            if j >= i:
+                rho_d[i*m:(i+1)*m, (j+1)*m:(j+2)*m] = list_of_ddls[i][j][0:m, m:m*2]
+                rho_d[(j+1)*m:(j+2)*m, i*m:(i+1)*m] = list_of_ddls[i][j][m:2*m, 0:m]
+    rho = rho_d
+    return rho
+    
+def two_body_guess(mol, monomers, dimers, bs):
+    monomer_dens = make_monomer_dens(monomers, bs)
+    m,m = monomer_dens[0].shape
+    a = len(monomer_dens)
+    list_of_ddls = make_list_of_ddls(dimers, bs)
+    z_rho = make_z_rho(m, a)
+    ei_list = make_ei_list(list_of_ddls)
+    i_list = fill_ei_list(list_of_ddls, ei_list, m)
+    overlap = add_interacts(i_list, m)
+    diags = compute_diags(overlap, monomer_dens, a)
+    rho_d = rho_diagonal(diags, z_rho, m)
+    rho = complete_rho(list_of_ddls, rho_d, m)
+    return rho
+
+bs = "aug-cc-pvdz"
+mol = [("O", -0.167787, 1.645761, 0.108747),
+       ("H", 0.613411, 1.102620, 0.113724),
+       ("H", -0.093821, 2.209720, -0.643619),
+       ("O", 1.517569, -0.667424, -0.080674),
+       ("H", 1.989645, -1.098799, 0.612047),
+       ("H", 0.668397, -1.091798, -0.139744),
+       ("O", -1.350388, -0.964879, -0.092208),
+       ("H", -1.908991, -1.211298, 0.626207),
+       ("H", -1.263787, -0.018107, -0.055536)]
+
+monomers = [mol[0:3], mol[3:6], mol[6:9]]
+ms = monomers
+dimers = [[(ms[0]+ms[1]), (ms[0]+ms[2])], [(ms[0]+ms[1]), (ms[1]+ms[2])], [(ms[0]+ms[2]), (ms[1]+ms[2])]]
+
+H2OA = density(mol[0:3], bs)
+H2OB = density(mol[3:6], bs)
+H2OC = density(mol[6:9], bs)
+H4O2AB = density(mol[0:6], bs)
+H4O2AC = density((mol[0:3]+mol[6:9]), bs)
+H4O2BC = density(mol[3:9], bs)
+
+m,m = H2OA.shape
+n = m*2
+A = np.zeros((m*3, m*3))
+
+A[0:m, 0:m] = H4O2AB[0:m, 0:m]+H4O2AC[0:m, 0:m]-H2OA
+A[m:n, m:n] = H4O2AB[m:n, m:n]+H4O2BC[0:m, 0:m]-H2OB
+A[n:m*3, n:m*3] = H4O2AC[m:n, m:n]+H4O2BC[m:n, m:n]-H2OC
+A[0:m, m:n] = H4O2AB[0:m, m:n]
+A[m:n, 0:m] = H4O2AB[m:n, 0:m]
+A[0:m, n:m*3] = H4O2AC[0:m, m:n]
+A[n:m*3, 0:m] = H4O2AC[m:n, 0:m]
+A[m:n, n:m*3] = H4O2BC[0:m, m:n]
+A[n:m*3, m:n] = H4O2BC[m:n, 0:m]
+
+rhoof = A
+
+if __name__=="__main__":
+    mol = [("O", -0.167787, 1.645761, 0.108747),
+       ("H", 0.613411, 1.102620, 0.113724),
+       ("H", -0.093821, 2.209720, -0.643619),
+       ("O", 1.517569, -0.667424, -0.080674),
+       ("H", 1.989645, -1.098799, 0.612047),
+       ("H", 0.668397, -1.091798, -0.139744),
+       ("O", -1.350388, -0.964879, -0.092208),
+       ("H", -1.908991, -1.211298, 0.626207),
+       ("H", -1.263787, -0.018107, -0.055536)]
+
+    bs = "aug-cc-pvdz"
+    monomers = [mol[0:3], mol[3:6], mol[6:9]]
+    ms = monomers
+    dimers = [[(ms[0]+ms[1]), (ms[0]+ms[2])], [(ms[0]+ms[1]), (ms[1]+ms[2])], [(ms[0]+ms[2]), (ms[1]+ms[2])]]
+    
+    tbg = two_body_guess(mol, monomers, dimers, bs)
+    #nitr = run_scf(mol, bs, tbg)
+    #print(tbg)
+
+
+class MyTest(unittest.TestCase):
+    def test(self):
+        self.assertEqual(two_body_guess(mol, monomers, dimers, bs), rhoof)
+        
+if __name__ == '__main__':
+    unittest.main()
```

Comment:
Could be obvious, but:

E
======================================================================
ERROR: /home/jacob/ (unittest.loader._FailedTest)
----------------------------------------------------------------------
AttributeError: module '__main__' has no attribute '/home/jacob/'

----------------------------------------------------------------------
Ran 1 test in 0.001s

FAILED (errors=1)

Reply:
Try putting the unit test in a separate file.

Reply:
Still no change in the above status after changing the unittest code.

Reply:
What you see here is exactly what I have on a different file.

Reply:
I made a new file, different from the one used to develop the code, specifically for testing as you suggested. This whole script is on that test file.

Reply:
Did you forget to check some other files in? To clarify, typically you define your functions in one file, your unit tests in another, and then you collect your data in a third. This separates your concerns (implementation, testing, and application) making it easier to do one without affecting the others.

Reply:
By 'checking in' do you mean import? Yeah I put the 2b-scf functions in a different file and imported it on the test file. I also put the test trimer script on its own file, and imported that.

Reply:
Checking in refers to adding the files to the repo.

Reply:
Right now I have them in their own .py files, but I have not added them to my local api.